pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -892,9 +894,12 @@ func (s *DiscoveryServer) startPush(version string, push *model.PushContext, ful
 			}:
 				client.LastPush = time.Now()
 				client.LastPushFailure = timeZero
+				if !timer.Stop() {
+					<-timer.C
```

Comment:
`Stop(): It returns true if the call stops the timer, false if the timer has already expired or been stopped`

Why do we receive on the timer channel when its already expired? Seems like we just need a `timer.Stop()`

Reply:
The documentation states this https://golang.org/pkg/time/#Timer.Stop:
```
To prevent a timer created with NewTimer from firing after a call to Stop, check the return value and drain the channel. For example, assuming the program has not received from t.C already:
```

Reply:
The doc says this can prevent read from the channel `timer.C`. If we make sure no one is reading from this channel, then no need call `<-timer.C` here. The golang will gc this timer.

Reply:
I don't think we can guarantee this is true given the case will read from C below?